lib/components/atoms/Pagination/Pagination.js

```diff
@@ -0,0 +1,93 @@
+// @flow
+/**
+ *
+ * Pagination
+ *
+ */
+import React from 'react';
+import styled from 'styled-components';
+
+import styles from './Pagination.style';
+import type { Props } from './types';
+import Button from '../Button';
+
+const Pagination = ({
+  postsPerPage,
+  totalPosts,
+  paginate,
+  className,
+  firstLabel,
+  lastLabel,
+  firstPost,
+  lastPost,
+  nextLabel,
+  previousLabel,
+  currentPage,
+}: Props): Node => {
+  const pageNumbers = [];
+  let i = 0;
+
+  do {
+    i += 1;
+    pageNumbers.push(i);
+  } while (i <= Math.ceil(totalPosts / postsPerPage));
+
+  const disabledFirstNdPrevious = firstPost <= 1;
+  const disabledLast = lastPost === 1;
+  const disabledNext = currentPage >= pageNumbers.length;
+
+  return (
+    <nav className={className}>
+      <ul className="pagination">
+        <li className="first">
+          <Button
+            className="btn-first"
+            disabled={disabledFirstNdPrevious}
+            onClick={() => paginate(1)}
+          >
+            {firstLabel}
+          </Button>
+        </li>
+        <li className="previous">
+          <Button disabled={disabledFirstNdPrevious} onClick={() => paginate(currentPage - 1)}>
+            {previousLabel}
+          </Button>
+        </li>
+        {pageNumbers.map(number => (
```

Comment:
Try break the component in smaller pieces, this could be a good small children component.  

<NumberedNav length={} activeNav={} />